app/backend/approaches/readretrieveread.py

```diff
@@ -44,29 +44,51 @@ class ReadRetrieveReadApproach(Approach):
 
     CognitiveSearchToolDescription = "useful for searching the Microsoft employee benefits information such as healthcare plans, retirement plans, etc."
 
-    def __init__(self, search_client: SearchClient, openai_deployment: str, sourcepage_field: str, content_field: str):
+    def __init__(self, search_client: SearchClient, openai_deployment: str, embedding_deployment: str, sourcepage_field: str, content_field: str):
         self.search_client = search_client
         self.openai_deployment = openai_deployment
+        self.embedding_deployment = embedding_deployment
         self.sourcepage_field = sourcepage_field
         self.content_field = content_field
 
-    def retrieve(self, q: str, overrides: dict[str, Any]) -> Any:
-        use_semantic_captions = True if overrides.get("semantic_captions") else False
+    def retrieve(self, query_text: str, overrides: dict[str, Any]) -> Any:
+        has_text = overrides.get("retrieval_mode") in ["text", "hybrid", None]
+        has_vector = overrides.get("retrieval_mode") in ["vectors", "hybrid", None]
+        use_semantic_captions = True if overrides.get("semantic_captions") and has_text else False
```

Comment:
I notice a lot of repeated code across the approaches now. Would it make sense to move any of it into a method on the Approach class? (These 3 lines, for examples)

Might help developers to see what's actually different across the approaches.

Reply:
The original goal of this demo was to be small, and each "approach" had a lot of redundancy because I thought people could take it, try the various approaches, and then pick one and delete the rest.

Clearly this codebase keeps growing, so yes, I some point we may want to consider refactoring to extract common parts. I'll keep this PR focused on introducing vector search, we can debate separately when we do a bit of refactoring.